lib/src/fnc/script/classes/duration.rs

```diff
@@ -13,31 +14,40 @@ pub mod duration {
 	#[quickjs(cloneable)]
 	pub struct Duration {
 		#[quickjs(hide)]
-		pub(crate) value: String,
+		pub(crate) value: Option<duration::Duration>,
 	}
 
 	impl Duration {
 		#[quickjs(constructor)]
 		pub fn new(value: String, args: Rest<Value>) -> Self {
 			Self {
-				value,
+				value: duration::Duration::try_from(value).ok(),
 			}
 		}
 		#[quickjs(get)]
-		pub fn value(&self) -> &str {
-			&self.value
+		pub fn value(&self) -> String {
+			match &self.value {
+				Some(v) => v.to_raw(),
+				None => String::from("Invalid Duration"),
+			}
 		}
 		// Compare two Duration instances
 		pub fn is(a: &Duration, b: &Duration, args: Rest<Value>) -> bool {
 			a.value == b.value
 		}
```

Comment:
The semantics of this function have changed slightly i.e. two arbitrary invalid durations will lead to it returning `true` because they are both `None` instead of `false` because they are both arbitrary `String`'s.

Not sure if/where this function is used or if this change matters, but pointing it out just in case.